hybrid_identity.py

```diff
@@ -60,7 +60,8 @@ def authenticate(self, user_id, password):
             assert utils.check_password(password, user_ref['password'])
         except TypeError:
             raise AssertionError('Invalid user / password')
-        except KeyError:  # if it doesn't have a password, it must be LDAP
+        except (ValueError, KeyError):
+            # if it doesn't have a password, it must be LDAP
```

Comment:
Hm, under which conditions does this throw a ```ValueError```? Isn't ```KeyError``` already enough to catch the case where user_ref has no password?